retis/src/core/filters/meta/filter.rs

```diff
@@ -268,160 +569,558 @@ impl MetaOp {
             };
         }
 
-        if mask > 0 {
-            if lop.is_ptr() || (lop.is_num() && !lop.is_signed()) {
-                lop.mask = mask;
+        if field.mask > 0 {
+            if ctx.load.is_ptr() || (ctx.load.is_num() && !ctx.load.is_signed()) {
+                ctx.load.mask = field.mask;
             } else {
                 bail!("mask is only supported for pointers and unsigned numeric members.");
             }
         }
 
-        lop.bf_size = u8::try_from(bfs)?;
-        lop.offt = u16::try_from(offt)?;
-
-        if bfs == 0 {
-            lop.offt /= 8;
-        }
-
-        Ok(op)
-    }
+        ctx.load.offt = u16::try_from(self.offt)?;
 
-    fn emit_target(lmo: &MetaLoad, rval: Rval, cmp_op: MetaCmp) -> Result<MetaOp> {
-        let mut op: MetaOp = MetaOp::new();
-        let top = op.target_ref_mut();
-
-        if lmo.is_ptr() || lmo.nmemb > 0 {
-            if cmp_op != MetaCmp::Eq && cmp_op != MetaCmp::Ne {
+        if ctx.load.is_ptr() || ctx.load.nmemb > 0 {
+            if rel_op != RelOp::Eq && rel_op != RelOp::Ne {
                 bail!(
                     "wrong comparison operator. Only '{}' and '{}' are supported for strings.",
-                    MetaCmp::Eq,
-                    MetaCmp::Ne
+                    RelOp::Eq,
+                    RelOp::Ne
                 );
             }
 
-            if let Rval::Str(val) = rval {
-                let rval_len = val.len();
-                let md = &mut top.md;
-                if rval_len >= md.len() {
-                    bail!("invalid rval size (max {}).", md.len() - 1);
+            if let Rhs::Str(val) = rval {
+                // lenght including '\0'
+                let rval_len = val.len() + 1;
+                let md = &mut ctx.target.md;
+
+                ctx.target.sz = rval_len
+                    .try_into()
+                    .map_err(|_| anyhow!("invalid rval size (max {})", md.len() - 1))?;
+                md[..val.len()].copy_from_slice(val.as_bytes());
+                if ctx.load.nmemb == 0 {
```

Comment:
Not sure to get when `nmemb` would be 0 and valid? Also couldn't that trigger OOB loads?

Reply:
off the top of my head is the `char *` case. Can't determine statically the lval size, so let's just probe_read (safe but could be incorrect, up to the user) rval_len bytes.

Reply:
The case is that, no doubt.
The problem is finding an actual `char *` in the `sk_buff` :)